src/librustc_lint/builtin.rs

```diff
@@ -691,6 +691,17 @@ impl EarlyLintPass for DeprecatedAttr {
                 return;
             }
         }
+        if attr.check_name(sym::no_start) || attr.check_name(sym::crate_id) {
+            let msg = format!("use of deprecated attribute `{}`: no longer used.", attr.path);
+            let mut err = cx.struct_span_lint(DEPRECATED, attr.span, &msg);
+            err.span_suggestion_short(
+                attr.span,
+                "remove this attribute",
+                String::new(),
+                Applicability::MachineApplicable
+            );
+            err.emit();
```

Comment:
This is somewhat similar to the previous bit but is sufficiently different that I didn't see an easy way to unify the two. I don't personally think it's likely anyone will hit this deprecation notice today due to how long it's been since these attributes were used by the compiler, so I'd like to avoid spending time making it super nice if possible.

Reply:
This should do it:
```rust
fn lint_deprecated_attr(cx: &EarlyContext<'_>, attr: &Attribute, msg: &str, suggestion: Option<&str>) {
    cx.struct_span_lint(DEPRECATED, attr.span, &msg)
        .span_suggestion_short(
            attr.span,
            suggestion.unwrap_or("remove this attribute"),
            String::new(),
            Applicability::MachineApplicable
        )
        .emit();
}
```

Reply:
That's a lot of arguments to the function, but it does seem marginally better. I'll make this change once we figure out whether we're keeping the lint or cratering removal.

Reply:
You could use a lambda to get rid of two of those arguments.